dags/mozaggregator_prerelease_new.py

```diff
@@ -0,0 +1,181 @@
+
+is_dev = os.environ.get("DEPLOY_ENVIRONMENT") == "dev"
+client_email = (
+    keyfile["client_email"]
+    if is_dev
+    else "dataproc-runner-prod@airflow-dataproc.iam.gserviceaccount.com"
+)
+artifact_bucket = (
+    "{}-dataproc-artifacts".format(project_id)
+    if is_dev
+    else "moz-fx-data-prod-airflow-dataproc-artifacts"
+)
+storage_bucket = (
+    "{}-dataproc-scratch".format(project_id)
+    if is_dev
+    else "moz-fx-data-prod-dataproc-scratch"
+)
+
+
+prerelease_telemetry_aggregate_view_dataproc = SubDagOperator(
+    task_id=task_id,
+    dag=dag,
+    subdag=moz_dataproc_pyspark_runner(
+        parent_dag_name=dag.dag_id,
+        dag_name=task_id,
+        job_name="prerelease_aggregates_new",
+        cluster_name="prerelease-telemetry-aggregates-new-{{ ds_nodash }}",
+        idle_delete_ttl="600",
+        num_workers=10,
+        worker_machine_type="n1-standard-8",
+        init_actions_uris=[
+            "gs://dataproc-initialization-actions/python/pip-install.sh"
+        ],
+        additional_properties={
+            "spark:spark.jars": "gs://spark-lib/bigquery/spark-bigquery-latest.jar",
+            "spark:spark.jars.packages": "org.apache.spark:spark-avro_2.11:2.4.4",
+        },
+        additional_metadata={
+            "PIP_PACKAGES": "git+https://github.com/mozilla/python_mozaggregator.git"
+        },
+        python_driver_code="gs://{}/jobs/mozaggregator_runner.py".format(
+            artifact_bucket
+        ),
+        py_args=[
+            "aggregator",
+            "--date",
+            "{{ ds_nodash }}",
+            "--channels",
+            "nightly,aurora,beta",
+            "--postgres-db",
+            "telemetry",
+            "--postgres-user",
+            "root",
+            "--postgres-pass",
+            "{{ var.value.mozaggregator_lean_postgres_pass }}",
+            "--postgres-host",
+            "{{ var.value.mozaggregator_lean_postgres_host }}",
+            "--postgres-ro-host",
+            "{{ var.value.mozaggregator_lean_postgres_ro_host }}",
+            "--num-partitions",
+            str(10 * 32),
+        ]
+        + (
+            ["--source", "bigquery", "--project-id", "moz-fx-data-shared-prod"]
+            if not EXPORT_TO_AVRO
+            else [
+                "--source",
+                "avro",
+                "--avro-prefix",
+                "gs://moz-fx-data-derived-datasets-parquet-tmp/avro/mozaggregator/prerelease-new/moz-fx-data-shared-prod",
+            ]
+        ),
+        gcp_conn_id=gcp_conn.gcp_conn_id,
+        service_account=client_email,
+        artifact_bucket=artifact_bucket,
+        storage_bucket=storage_bucket,
+        default_args=subdag_args,
+    ),
+)
+
+
+# export to avro, if necessary
+if EXPORT_TO_AVRO:
+    gke_command(
+        task_id="export_main_avro",
+        cmds=["bash"],
+        command=[
+            "bin/export-avro.sh",
+            "moz-fx-data-shared-prod",
+            "moz-fx-data-shared-prod:analysis",
```

Comment:
I hope nothing weird happens here, because the same table gets reused for exports. This takes 10 minutes, so hopefully the new job and the old job dont run exactly in parallel.

Reply:
With respect to this comment, the way to avoid table re-use is to specify a different dataset, possibly `moz-fx-data-shared-prod:mozaggregator-backfill-tmp`, which does not overlap with `analysis`.

Reply:
See: https://github.com/mozilla/python_mozaggregator/blob/207068afed33c592f1d49904b16881910459b878/bin/export-avro.sh#L59-L61